actioncable/test/client_test.rb

```diff
@@ -198,4 +198,22 @@ def test_disappearing_client
       c.close # disappear before read
     end
   end
+
+  def test_unsubscribe_client
+    with_puma_server do |port, app|
+      c = faye_client(port)
+      c.send_message command: 'subscribe', identifier: JSON.dump(channel: 'EchoChannel')
+      assert_equal({"identifier"=>"{\"channel\":\"EchoChannel\"}", "type"=>"confirm_subscription"}, c.read_message)
+      assert_equal(1, app.connections.count)
+      assert_equal(1, app.connections.first.subscriptions.method(:subscriptions).call.count)
+
+      channel = app.connections.first.subscriptions.method(:subscriptions).call.first[1]
+      channel.expects(:unsubscribed).returns('Goodbye from EchoChannel!')
```

Comment:
Yes -- I want to make sure that the `unsubscribed` method is properly called and handled.